### DISCUSSION TITLE
Tamagotchi - Why use private variables over object key value pair in each component?

### DISCUSSION THREAD

**Discussion by jrobinson0529** [nss-evening-cohort-14/evening-client-side#141] [Discuss a Topic]
As the title states I am confused on why it says to use a private variable instead of setting a property on a pet object for each point counter. I tried searching for an answer on google but the question is too niche.

**Comment by drteresavasquez** [comment]
Can you point out the specific thing you're talking about? I have no idea what you're referencing. Poe favoe 🤣

**Comment by jrobinson0529** [comment]
Sorry about that. This is the section I am referencing.
Eat
The Eat component should appear in the upper left quadrant. It should have a private variable called full. This variable should be set to 100 on page load. This module should have 2 buttons. One should be a healthy food that adds 10 to full when pressed. The other button should be an unhealthy food that subtracts 3 from full when pushed.

The quadrant should display the name, fullness score, and the two buttons. Each time a button is clicked the quadrant should be re-drawn to show the correct fullness score. The Full score should NEVER be larger than 100.

Play
The Play component should appear in the upper right quadrant. It should have a private variable called fun. This variable should be set to 50 on page load. This module should have 2 buttons. One should be a super fun activity that adds 50 to fun when pressed. The other button should be a slightly fun activity that adds 2 to fun when pushed.

The quadrant should display the name, fun score, and the two buttons. Each time a button is clicked the quadrant should be re-drawn to show the correct fun score. The Fun score should NEVER be larger than 100.

Fight
The Fight component should appear in the lower left quadrant. It should have a private variable called strength. This variable should be set to 100 on page load. This module should have 2 buttons. One should be for running away (bravely) that adds 1 to strength when pressed. The other button should be for committing violence and should subtract 10 from strength when pushed.

The quadrant should display the name, strength score, and the two buttons. Each time a button is clicked the quadrant should be re-drawn to show the correct strength score. The Strength score should NEVER be larger than 100.

Sleep
The Sleep component should appear in the lower right quadrant. It should have a private variable called energy. This variable should be set to 50 on page load. This module should have 2 buttons. One should be for a nap and should add 50 to energy when pressed. The other button should be for a deep slumber and add 60 to sleep when pushed.

The quadrant should display the name, energy score, and the two buttons. Each time a button is clicked the quadrant should be re-drawn to show the correct energy score. The Energy score should NEVER be larger than 100.

**Comment by jrobinson0529** [comment]
It says each component has a "private variable" and I am confused by the wording. Why would we not just it as object keys instead?

**Comment by drteresavasquez** [comment]
Oh ok. Because that’s what the requirements call for 😊. AND since each quadrant will likely have its own module, that data should be private to that module. Same as we having separate data files for each type of data.

**Reply by jrobinson0529** [reply]
I guess I am just having trouble understanding how the data will interact with each other when you start crossing modules with the manipulation. IE Food goes up and strength goes down from food coma. I'm sure I will get it once I dive deeper in..

**Reply by drteresavasquez** [reply]
I see! It will likely be a helper function that you create that pulls in those modules that you can call into your view. So in helpers, you have data and you can have modules that you create to combine those and pull them where needed. 

It will take some planning but it's good you're thinking about this!

**Reply by jrobinson0529** [reply]
I appreciate the help. Definitely looking forward to diving into this one.

**Reply by drteresavasquez** [reply]
It's a very fun project!

**Comment by katherinevfry** [comment]
I had an issue with this too! To make my variable private, I declared it outside the function I'm exporting. For instance:

```
let full = 100;

const eatFood = () => {
// do something with the full variable
};
```

**Comment by drteresavasquez** [comment]
To further explain this concept, here is an example file for `eat`

```javascript
let full = 100;

const healthy = () => {
   ...
};

const unhealthy = () => {
  ...
};

// This function grabs the full variable and returns the value at the time of the function call
const getEatTotal = () => full; 

export { healthy, unhealthy, getEatTotal };
```
